src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -543,14 +544,22 @@ const AppWrapper = (props: { children: ReactNode }) => {
       await Agent.agent.devPreload();
     }
 
-    await loadDatabase();
     const { keriaConnectUrlRecord } = await loadCacheBasicStorage();
 
     // Ensure online/offline callback setup before connecting to KERIA
     setupEventServiceCallbacks();
 
     if (keriaConnectUrlRecord) {
       try {
+        if (keriaConnectUrlRecord?.content?.url) {
+          const recoveryStatus = await Agent.agent.basicStorage.findById(
```

Comment:
What happens if KERIA is down when we are mid-way through recovering? It seems to me that we will call `connect` and carry on like nothing happened, and we won't have called `loadDatabase` either. Please actually test this scenarios by throwing an error in your local testing.